Core/GraphicsInterops/Include/Shared/Babylon/GraphicsInterops/ExternalTexture.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+#include <Babylon/JsRuntime.h>
+#include <Babylon/Graphics/Platform.h>
+#include <Babylon/Graphics/RendererType.h>
+#include <Babylon/Graphics/Device.h>
+
+#include <future>
+#include <memory>
+
+namespace Babylon::Graphics
+{
+    class ExternalTexture
+    {
+    public:
+        ExternalTexture(TextureType nativeTexture);
+        ExternalTexture(ExternalTexture&&);
+        ~ExternalTexture();
+
+        Napi::Promise AddToContext(Napi::Env&);
+
+        uint32_t GetWight();
+        uint32_t GetHeight();
+        uint32_t GetFormat();
+
+    private:
+        struct Impl;
+        std::unique_ptr<Impl> m_impl{};
+
+        ExternalTexture(std::unique_ptr<Impl>);
```

Comment:
If I remember correctly, this was here to be used by a static factory. With the static factory gone, do we need this private constructor?

Reply:
You are right @syntheticmagus , the way we are currently doing things do not require this indirection. I have completely removed ExternalTextureImpl and made everything part of ExternalTexture.